src/components/interpolatedColorLegend.ts

```diff
@@ -0,0 +1,249 @@
+     * associated Scale.InterpolatedColor sampled at various points. Two labels
+     * show the maximum and minimum values of the Scale.InterpolatedColor.
+     *
+     * @constructor
+     * @param {Scale.InterpolatedColor} interpolatedColorScale
+     * @param {string} orientation (horizontal/left/right).
+     * @param {Formatter} The labels are formatted using this function.
+     */
+    constructor(interpolatedColorScale: Scale.InterpolatedColor, orientation = "horizontal", formatter = Formatters.general()) {
+      super();
+      if (interpolatedColorScale == null ) {
+        throw new Error("InterpolatedColorLegend requires a interpolatedColorScale");
+      }
+      this._scale = interpolatedColorScale;
+      this._formatter = formatter;
+      this._orientation = InterpolatedColorLegend.validateOrientation(orientation);
+
+      this._fixedWidthFlag = true;
+      this._fixedHeightFlag = true;
+    }
+
+    /**
+     * Gets the current formatter on the InterpolatedColorLegend.
+     *
+     * @returns {Formatter} The current Formatter.
+     */
+    public formatter(): Formatter;
+    /**
+     * Sets the current formatter on the InterpolatedColorLegend.
+     *
+     * @param {Formatter} formatter If provided, data will be passed though `formatter(data)`.
+     * @returns {InterpolatedColorLegend} The calling InterpolatedColorLegend.
+     */
+    public formatter(formatter: Formatter): InterpolatedColorLegend;
+    public formatter(formatter?: Formatter): any {
+      if (formatter === undefined) {
+        return this._formatter;
+      }
+      this._formatter = formatter;
+      this._invalidateLayout();
+      return this;
+    }
+
+    private static validateOrientation(orientation: string) {
+      orientation = orientation.toLowerCase();
+      if (orientation === "horizontal" || orientation === "left" || orientation === "right") {
+        return orientation;
+      } else {
+        throw new Error("\"" + orientation + "\" is not a valid orientation for InterpolatedColorLegend");
+      }
+    }
+
+    /**
+     * Gets the orientation of the InterpolatedColorLegend.
+     *
+     * @returns {string} The current orientation.
+     */
+    public orient(): string;
+    /**
+     * Sets the orientation of the InterpolatedColorLegend.
+     *
+     * @param {string} newOrientation The desired orientation (horizontal/left/right).
+     *
+     * @returns {InterpolatedColorLegend} The calling InterpolatedColorLegend.
+     */
+    public orient(newOrientation: string): InterpolatedColorLegend;
+    public orient(newOrientation?: string): any {
+      if (newOrientation == null) {
+        return this._orientation;
+      } else {
+        this._orientation = InterpolatedColorLegend.validateOrientation(newOrientation);
+        this._invalidateLayout();
+        return this;
+      }
+    }
+
+    protected _setup() {
+      super._setup();
+      this.classed("legend", true).classed("interpolated-color-legend", true);
```

Comment:
Also, not sure how much it matters, but if the user does not want a legend class stapled on, this will force it.